pkgs/development/tools/build-managers/muon/default.nix

```diff
@@ -0,0 +1,142 @@
+{ lib
+, stdenv
+, fetchFromSourcehut
+, fetchurl
+, curl
+, libarchive
+, libpkgconf
+, pkgconf
+, python3
+, samurai
+, scdoc
+, zlib
+, embedSamurai ? false
+, buildDocs ? true
+, static ? false
+}:
+
+let
+  # Courtesy of sternenseemann and FRidh
+  featureFlag = feature: flag:
+    "-D${feature}=${if flag then "enabled" else "disabled"}";
+  conditionFlag = condition: flag:
+    "-D${condition}=${if flag then "true" else "false"}";
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "muon"
+          + lib.optionalString embedSamurai "-embedded-samurai"
+          + lib.optionalString static "-static";
+  version = "unstable-2022-08-24";
+
+  src = fetchFromSourcehut {
+    owner = "~lattis";
+    repo = "muon";
+    rev = "54972be2a39cd98dc6a9a7f414f3c748d25e44b8";
+    hash = "sha256-pojks/uLiuiUu9UVO8XK/jMby/NipyG6Z3wOYYOUOJI=";
+  };
+
+  nativeBuildInputs = [
+    pkgconf
+    samurai
+  ]
+  ++ lib.optionals buildDocs [
+    (python3.withPackages (ps: [ ps.pyyaml ]))
+    scdoc
+  ];
+
+  buildInputs = [
+    curl
+    libarchive
+    libpkgconf
+    samurai
+    zlib
+  ];
+
+  strictDeps = true;
+
+  postUnpack = let
+    # URLs manually extracted from subprojects directory
+    meson-docs-wrap = fetchurl {
+      url = "https://mochiro.moe/wrap/meson-docs-0.63.0-40-g280f3423b.tar.gz";
+      hash = "sha256-/XSxroC8r1YXsabXloKRnbDiA8cO2Z7SHMmF+2ETkNA=";
+    };
+
+    samurai-wrap = fetchurl {
+      url = "https://mochiro.moe/wrap/samurai-1.2-27-g92711f7.tar.gz";
+      hash = "sha256-DqfOoGUtO34ECOod2FdqAbduLMJPGRxvbQOZ7515A2Y=";
+  };
+  in ''
+    pushd $sourceRoot/subprojects
+    ${lib.optionalString buildDocs "tar xvf ${meson-docs-wrap}"}
+    ${lib.optionalString embedSamurai "tar xvf ${samurai-wrap}"}
+    popd
+  '';
+
+  postPatch = ''
+    patchShebangs bootstrap.sh
+  ''
+  + lib.optionalString buildDocs ''
+    patchShebangs subprojects/meson-docs/docs/genrefman.py
+  '';
+
+  # tests try to access "~"
+  postConfigure = ''
+    export HOME=$(mktemp -d)
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    ./bootstrap.sh stage-1
+
+    ./stage-1/muon setup \
+      ${featureFlag "docs" buildDocs} \
+      ${featureFlag "samurai" embedSamurai} \
+      ${conditionFlag "static" static} \
+      stage-2
+    samu -C stage-2
+
+    stage-2/muon setup \
+      -Dprefix=$out \
+      ${featureFlag "docs" buildDocs} \
+      ${featureFlag "samurai" embedSamurai} \
+      ${conditionFlag "static" static} \
+      stage-3
+    samu -C stage-3
+
+    runHook postBuild
+  '';
+
+  # tests are failing because they don't find Python
+  doCheck = false;
+
+  checkPhase = ''
+    runHook preCheck
+
+    ./stage-3/muon -C stage-3 test
+
+    runHook postCheck
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    stage-3/muon -C stage-3 install
```

Comment:
This is ignoring nix flags like -j$NIX_BUILD_CORES.

Reply:
installation is just a copy, no compilation needed.